packages/@sanity/base/src/change-indicators/overlay/useResizeObserver.ts

```diff
@@ -0,0 +1,9 @@
+import {useEffect} from 'react'
+import {resizeObserver} from '../../util/resizeObserver'
+
+export function useResizeObserver(
+  element: HTMLDivElement,
+  onResize: (event: ResizeObserverEntry) => void
+): void {
+  useEffect(() => resizeObserver.observe(element, onResize), [element, onResize])
```

Comment:
old code but i think this is missing a clean up function to call `resizeObserver.disconnect`

Reply:
The `resizeObserver` does that internally in this case.